sdks/python/apache_beam/runners/direct/transform_evaluator.py

```diff
@@ -421,8 +424,12 @@ def process_element(self, element):
       main_output = list(self._outputs)[0]
       bundle = self._evaluation_context.create_bundle(main_output)
       for tv in event.timestamped_values:
-        bundle.output(
-            GlobalWindows.windowed_value(tv.value, timestamp=tv.timestamp))
+        # Unreify the value into the correct window.
+        try:
+          bundle.output(WindowedValue(**tv.value))
+        except TypeError:
+          bundle.output(
+              GlobalWindows.windowed_value(tv.value, timestamp=tv.timestamp))
```

Comment:
Question:
What happens if tv.value is a dict from the upstream pcollection? 
We are supposing that it's a dict containing the window, pane info, value, and tstamp OR if we get a  typeerror, then it's not a dict, it's only the element itself - but what if the element itself is a dict?

Reply:
Added a test for this